testdata/go/v3/monitoring/memcached-operator/monitoring/metricsdocs/metricsdocs.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/example/memcached-operator/monitoring"
+)
+
+// please run "make generate-metricsdocs" to run this tool and update metrics documentation
+const (
+	title      = "# Operator Metrics\n"
+	background = "This document aims to help users that are not familiar with metrics exposed by this operator.\n" +
+		"The metrics documentation is auto-generated by the utility tool \"monitoring/metricsdocs\" and reflects all of the metrics that are exposed by the operator.\n\n"
+
+	KVSpecificMetrics = "## Operator Metrics List\n"
+
+	opening = title +
+		background +
+		KVSpecificMetrics
+
+	// footer
+	footerHeading = "## Developing new metrics\n"
+	footerContent = "After developing new metrics or changing old ones, please run \"make generate-metricsdocs\" to regenerate this document.\n\n" +
+		"If you feel that the new metric doesn't follow these rules, please change \"monitoring/metricsdocs\" according to your needs.\n"
+
+	footer = footerHeading + footerContent
+)
+
+// TODO: scaffolding these helpers with operator-lib: https://github.com/operator-framework/operator-lib.
+
+// metricList contains the name, description, and type for each metric.
+func main() {
+	metricList := metricDescriptionListToMetricList(monitoring.ListMetrics())
+	sort.Sort(metricList)
+	writeToFile(metricList)
+}
+
+// writeToFile receives a list of metrics and writes them to a file.
+func writeToFile(metricsList metricList) {
+	fmt.Print(opening)
+	metricsList.writeOut()
+	fmt.Print(footer)
+}
```

Comment:
This function name is confusing, it says it writes them to a file but really this is printing to STDOUT.

Reply:
@everettraven Maybe change to `printToFile` or `createFile` WDYT?

Reply:
@avlitman this function does nothing with files, but rather writes to STDOUT. I think something like `writeToStdOut` conveys what the function is doing better.

Reply:
> @avlitman this function does nothing with files, but rather writes to STDOUT. I think something like `writeToStdOut` conveys what the function is doing better.

Yes maybe write **the** file. but I think your suggestion is good. fixing now

Reply:
Done.